Apps/WebClient/src/ClientApp/app/components/timeline/comment.vue

```diff
@@ -19,13 +54,48 @@
 <script lang="ts">
 import Vue from "vue";
 import UserComment from "@/models/userComment";
-import { Prop, Component } from "vue-property-decorator";
+import { Prop, Component, Emit, Watch } from "vue-property-decorator";
+import { faEllipsisV, IconDefinition } from "@fortawesome/free-solid-svg-icons";
 
 @Component
 export default class CommentComponent extends Vue {
-  @Prop({ default: "Default" }) comment!: UserComment;
+  @Prop() comment!: UserComment;
+  @Prop() editing!: UserComment;
+
+  private isEditMode: boolean = false;
+  private hasErrors: boolean = false;
+
   private formatDate(date: Date): string {
     return new Date(Date.parse(date + "Z")).toLocaleString();
   }
+
+  private get menuIcon(): IconDefinition {
+    return faEllipsisV;
+  }
+
+  @Watch("editing")
+  private onEditMode(): void {
+    this.editing.id === this.comment.id
+      ? (this.isEditMode = true)
+      : (this.isEditMode = false);
+  }
+
+  private deleteComment(): void {
+    this.onCommentDeleted(this.comment);
```

Comment:
if the action happens here, why bubble it up to the parent?

Reply:
See below